scene/gui/item_list.cpp

```diff
@@ -1204,7 +1210,7 @@ void ItemList::_notification(int p_what) {
 						max_len = size2.x;
 					}
 
-					Color txt_modulate = items[i].selected ? theme_cache.font_selected_color : (items[i].custom_fg != Color() ? items[i].custom_fg : theme_cache.font_color);
+					Color txt_modulate = items[i].selected ? theme_cache.font_selected_color : (hovered == i ? theme_cache.font_hovered_color : (items[i].custom_fg != Color() ? items[i].custom_fg : theme_cache.font_color));
```

Comment:
I think we're at a point where this line needs to be unrolled into a proper if-else.